### PR TITLE
[Merged by Bors] - feat(*/Pointwise): generalize some lemmas to `SMulZeroClass`

### PR DESCRIPTION
**Pull request by urkud**
---
@YaelDillies Are there any other files I should adjust simultaneously?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
This will conflict with #9199. Do you mind adapting your PR to match?

You should adjust `Order.Filter.Pointwise` too.

File: Mathlib/Data/Finset/Pointwise.lean [hunk 1]
```diff
@@ -2084,24 +2084,41 @@ lemma inv_op_smul_finset_distrib (a : α) (s : Finset α) : (op a • s)⁻¹ =
...
+variable [Zero β] [SMulZeroClass α β] [DecidableEq β] {s : Finset α} {t : Finset β}
+
+theorem smul_zero_subset (s : Finset α) : s • (0 : Finset β) ⊆ 0 := by simp [subset_iff, mem_smul]
+#align finset.smul_zero_subset Finset.smul_zero_subset
+
+theorem Nonempty.smul_zero (hs : s.Nonempty) : s • (0 : Finset β) = 0 :=
+  s.smul_zero_subset.antisymm <| by simpa [mem_smul] using hs
+#align finset.nonempty.smul_zero Finset.Nonempty.smul_zero
+
+theorem zero_mem_smul_finset {t : Finset β} {a : α} (h : (0 : β) ∈ t) : (0 : β) ∈ a • t :=
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Pointwise.lean:2098]:
```suggestion
theorem zero_mem_smul_finset (h : (0 : β) ∈ t) : (0 : β) ∈ a • t :=
```

File: Mathlib/Data/Finset/Pointwise.lean [hunk 2]
```diff
@@ -2084,24 +2084,41 @@ lemma inv_op_smul_finset_distrib (a : α) (s : Finset α) : (op a • s)⁻¹ =
...
+variable [Zero β] [SMulZeroClass α β] [DecidableEq β] {s : Finset α} {t : Finset β}
+
+theorem smul_zero_subset (s : Finset α) : s • (0 : Finset β) ⊆ 0 := by simp [subset_iff, mem_smul]
+#align finset.smul_zero_subset Finset.smul_zero_subset
+
+theorem Nonempty.smul_zero (hs : s.Nonempty) : s • (0 : Finset β) = 0 :=
+  s.smul_zero_subset.antisymm <| by simpa [mem_smul] using hs
+#align finset.nonempty.smul_zero Finset.Nonempty.smul_zero
+
+theorem zero_mem_smul_finset (h : (0 : β) ∈ t) : (0 : β) ∈ a • t :=
```
Comment (urkud) [inline Mathlib/Data/Finset/Pointwise.lean:2098]:
```suggestion
theorem zero_mem_smul_finset {a : α} (h : (0 : β) ∈ t) : (0 : β) ∈ a • t :=
```

File: Mathlib/Data/Finset/Pointwise.lean [hunk 3]
```diff
@@ -2084,24 +2084,41 @@ lemma inv_op_smul_finset_distrib (a : α) (s : Finset α) : (op a • s)⁻¹ =
...
+
+theorem Nonempty.smul_zero (hs : s.Nonempty) : s • (0 : Finset β) = 0 :=
+  s.smul_zero_subset.antisymm <| by simpa [mem_smul] using hs
+#align finset.nonempty.smul_zero Finset.Nonempty.smul_zero
+
+theorem zero_mem_smul_finset {a : α} (h : (0 : β) ∈ t) : (0 : β) ∈ a • t :=
+  mem_smul_finset.2 ⟨0, h, smul_zero _⟩
+#align finset.zero_mem_smul_finset Finset.zero_mem_smul_finset
+
+variable [Zero α] [NoZeroSMulDivisors α β] {a : α}
```
Comment (YaelDillies) [inline Mathlib/Data/Finset/Pointwise.lean:2102]:
```suggestion
variable [Zero β] [SMulZeroClass α β] [DecidableEq β] {s : Finset α} {t : Finset β} {a : α}

theorem smul_zero_subset (s : Finset α) : s • (0 : Finset β) ⊆ 0 := by simp [subset_iff, mem_smul]
#align finset.smul_zero_subset Finset.smul_zero_subset

theorem Nonempty.smul_zero (hs : s.Nonempty) : s • (0 : Finset β) = 0 :=
  s.smul_zero_subset.antisymm <| by simpa [mem_smul] using hs
#align finset.nonempty.smul_zero Finset.Nonempty.smul_zero

theorem zero_mem_smul_finset (h : (0 : β) ∈ t) : (0 : β) ∈ a • t :=
  mem_smul_finset.2 ⟨0, h, smul_zero _⟩
#align finset.zero_mem_smul_finset Finset.zero_mem_smul_finset

variable [Zero α] [NoZeroSMulDivisors α β]
```

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Pull request closed without merging**